src/Microsoft.Azure.ServiceBus/SessionReceivePump.cs

```diff
@@ -122,7 +122,13 @@ async Task SessionPumpTaskAsync()
                     concurrentSessionSemaphoreAcquired = true;
 
                     await this.maxPendingAcceptSessionsSemaphoreSlim.WaitAsync(this.pumpCancellationToken).ConfigureAwait(false);
-                    var session = await this.client.AcceptMessageSessionAsync().ConfigureAwait(false);
+
+                    IMessageSession session;
+                    if(string.IsNullOrEmpty(sessionHandlerOptions.SessionId))
+                        session = await this.client.AcceptMessageSessionAsync().ConfigureAwait(false);
+                    else
+                        session = await this.client.AcceptMessageSessionAsync(sessionHandlerOptions.SessionId).ConfigureAwait(false);
```

Comment:
This means Session Handler will be pinned to the `SessionId` and process only one sessions for its entire lifetime. Is that the intent @alberthoekstra?

Reply:
Yes, that’s exactly what I want to happen.

Reply:
Changed t. Check latest commit.

Reply:
In that case I'd suggest to elaborate in more details under the [general description](https://github.com/Azure/azure-service-bus-dotnet/pull/624#issue-239293202) to provide team with an idea what this PR is trying to accomplish and how it would work.

Reply:
I’ve changed the description. Is this enough?